spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/CloudSqlEnvironmentPostProcessor.java

```diff
@@ -170,14 +170,6 @@ private void setCredentialsFileProperty(Resource credentialsLocation) {
 			// A resource might not be in the filesystem, but the Cloud SQL credential must.
 			File credentialsLocationFile = credentialsLocation.getFile();
 
-			// This should happen if the Spring resource isn't in the filesystem, but a URL,
-			// classpath file, etc.
-			if (credentialsLocationFile == null) {
```

Comment:
Are we saying that it actually cannot be null?

Reply:
It in fact cannot be null because there is a null check at the (one and only) call site.

Reply:
The null check there is for getLocation() not getFile()